Survival Analysis (#27)
mlsa-book/MLSA: book/survival.qmd

$$

The hazard function is interpreted as the instantaneous risk of death given that the observation has survived up until that point.
Note this is not a probability as $h_Y$ can be greater than one.

The cumulative hazard function (chf) can be derived from the hazard function by
$$
H_Y: \NNReals \rightarrow \NNReals; \quad
(\tau) \mapsto \int^\tau_0 h_Y(u) \ du
$$

The cumulative hazard function relates simply to the survival function by
$$
H_Y(\tau) = \int^\tau_0 h_Y(u) \ du = \int^\tau_0 \frac{f_Y(u)}{S_Y(u)} \ du = \int^\tau_0 -\frac{S'_Y(u)}{S_Y(u)} \ du = -\log(S_Y(\tau))
$$

These last relationships are particularly important, as many methods estimate the hazard rate, which is then used to calculate the cumulative hazard and survival probability
$$S_Y(\tau) = \exp(-H_Y(\tau)) = \exp\left(-\int_0^\tau h_Y(u)\ du\right).$${#eq-surv-haz}

From now on we will often refer to these quantities as $S$, $h$, etc. rather than $S_Y$, $h_Y$ for convenience, unless it's necessary to use the subscript to avoid confusion.

Normally, these quantities could be estimated using standard techniques like regression modeling (estimation of parameters of an assumed distribution).

> However, in contrast to standard settings, $Y$ is only observed partially, due to different types of censoring and truncation describeb below.

Author: anything in particular you don't like about the previous version?
In the edit, the statement is not true, it can be estimated via MLE, but the likelihood is non-standard.

Note that for this type of data you need an additional variable in order to differentiate between left- and right-censoring as well as exact event times.
In practice this is often solved by including two columns that store time, such that left-censored data is given by $(-\infty, A_i)$, while right-censored data is denoted by $(A_i, \infty)$.
In this convention, actual event times are denoted by $(T_i, T_i)$.

#### Interval-censoring {.unnumbered .unlisted}

Interval-censoring occurs when the event takes place in some interval within the study period, but the exact time of event is unknown.
This often occurs in data resulting from regular or irregular check-ups, as it often occurs in electronic health record data.
For example, patients might get periodic tests for skin cancer.
Some patients might check once a year, for others the intervals between checks might be longer.
If skin cancer is detected, one only knows that it occured between the last and current check up, but not the exact time.

Formally, consider $L_i$ the subject-specific check-up time and $R_i$ the current check-up time.
Then we know that if an event occured $T_i = Y_i \in (L_i, R_i]$.

> If no event occured then $Y_i > T_i = C_i = R_i$.

Author: I've simplified this, but don't think we need repeated measures. While check ups occur regularly in different intervals. We always only need the last interval, because then the event either occurred ($Y_i \in (L_i, R_i]$) or didn't, so $R_i = C_i$. We don't need the info about previous intervals for interval censoring definition.

While truncation also occurs in non time-to-event settings, it is quit common in survival analysis, especially left-truncation.

In general, while censored observations have incomplete information about the time-to-event, they are still part of the data set.
Truncation on the other hand often leads to observations not entering the data set (at least not at time 0).
This will usually introduce bias that needs to be accounted for.

#### Left-truncation {.unnumbered .unlisted}

Left-truncation often occurs when study participation is conditional of the occurence of another event. This is easiest explained through some examples:

In one famous study (reference), the interest was in infant survival during the first 365 days after birth based on whether the mother was alive.
The study is from the 18the century, when childhood and maternal mortality were compartively high.
Infants entered the study when the mother died, and two other infants with same age and other features were matched into the study, except that their mothers were still alive.

> The age of the infant at entrance into the study therefore signifies a so-called *left-truncation event*, as infants who die before their mothers never enter the data and the *left-truncation time* is given by the infants' age at which the mother died.

Author: I now added a figure + more description for right-truncation. Will add similar content for left-truncation later.

Therefore a survival model is unlikely to distinguish the censoring event (aneurysm) from the event of interest (stroke) and will confuse predictions.
In practice, the majority of models and measures assume that censoring is conditionally event-independent and hence censoring can be predicted by the covariates whilst not directly depending on the event.
For example if studying the survival time of ill pregnant patients in hospital, then dropping out of the study due to pregnancy is clearly dependent on how many weeks pregnant the patient is when the study starts (for the sake of argument assume no early/late pregnancy due to illness).

### Censoring vs. Truncation {#sec-truncation}
While sometimes confused or missnamed, it is very important to differentiate
between censoring and truncation, as the way the are handled and the methods suitable to handle respective survival tasks differs substantially.
While truncation also occurs in non time-to-event settings, it is quit common in survival analysis, especially left-truncation.

In general, while censored observations have incomplete information about the time-to-event, they are still part of the data set.

> Truncation on the other hand often leads to observations not entering the data set (at least not at time 0).

Author: I think there are two cases
1) subjects don't enter the data set at all (e.g. subjects die before entering retirement home). Then left-truncation has to be accounted for in order to estimate age at death distribution of subjects who do enter retirement home
2) in multi-state processes, subjects enter the risk set for secondary, tertiary, etc. transition at different time points (or not at all, when they remain in state 0, 1, 2, etc.), so the subjects are in the data set, but when estimating the transition at any given time point, they might not be in the risk set for this transition.

Probably worth discussing left-truncation a bit more

In one famous study (reference), the interest was in infant survival during the first 365 days after birth based on whether the mother was alive.
The study is from the 18the century, when childhood and maternal mortality were compartively high.
Infants entered the study when the mother died, and two other infants with same age and other features were matched into the study, except that their mothers were still alive.
The age of the infant at entrance into the study therefore signifies a so-called *left-truncation event*, as infants who die before their mothers never enter the data and the *left-truncation time* is given by the infants' age at which the mother died.

More formally, let $T^l_i$ the subject-specific left truncation time.
Then we only observe subjects with $Y_i > T^l_i$ or $C_i > T^l_i$. Subjects with $Y_i < T^l_i$ never enter the data.

Notably, left-truncation also plays an important role when modeling recurrent events or multi-state data, as the data generating process induces left-truncation (@sec-eha).

#### Right-truncation {.unnumbered .unlisted}

Right-truncation mostly occurs in analyses based on registry data.

> When sampling the registry for events, some subjects might already have experienced the event, but the registry data at a given time-point does not reflect this yet.

Author: I've updated this paragraph

A popular alternative is therefore to estimate the *restricted mean survival time* (RMST) [@han.restricted.2022; @andersen.regression.2004].

<!-- FIXME add example -->

#### Prognostic Index {.unnumbered .unlisted}
Given covariates, $\XX \in \Reals^{n \times p}$, and a vector of model coefficients, $\bbeta \in \Reals^p$, the linear predictor is defined by $\boldsymbol{\eta}:= \xx^\top\bbeta \in \Reals^n$.
The prognostic index is a term that is often used in survival analysis papers that usually refers to some transformation (possibly identity), $\phi$, on the linear predictor, $\phi(\eta)$.
Assuming a predictive function (for survival time, risk, or distribution defining function) of the form $g(\varphi)\phi(\eta)$, for some function $g$ and variables $\varphi$ where $g(\varphi)$ is constant for all observations (e.g. Cox PH (@sec-surv-models-crank)), then predictions of $\eta$ are a special case of predicting a relative risk, as predictions of $\phi(\eta)$ if $\phi$ is rank preserving.

> A higher prognostic index may imply a higher or lower risk of event, dependent on the model structure.

Reviewer: ```suggestion
A higher prognostic index may imply a higher or lower risk of event, dependent on the model structure.
As stated above, if the prognostic index is used for a rank prediction, then this book always assumes "higher value higher risk".
```

<!-- FIXME example needed -->

#### Survival Distribution {.unnumbered .unlisted}

> Predicting a survival distribution refers specifically to predicting the distribution of an individual patient's survival time, i.e. modelling the distribution of the event occurring over $\NNReals$.

Reviewer: ```suggestion
Predicting a survival distribution refers specifically to predicting the distribution of an individual subject's survival time, i.e., modelling the distribution of the event occurring over $\NNReals$.
```

Therefore this is seen as the probabilistic analogue to the deterministic time-to-event prediction, these definitions are motivated by similar terminology in machine learning regression problems (@sec-surv-setml).

> The above three prediction types can all be derived from a probabilistic survival distribution prediction (@sec-car).

Reviewer: ```suggestion
```

Distributional prediction can in theory target any of the quantities introduced in @sec-distributions, however, in survival context $S(t)$ and $h(t)$ are the most common.
Hazard based approaches are particularly relevant for non- and semi-parametric estimation of the distribution, where no assumptions are made about the underlying distribution of event times.

## Survival Analysis Task {#sec-surv-setmltask}

> 

Reviewer: Below remove parantheses around the section reference

## Survival Task

::: {#cnj-task-surv}
Let $(X,T,\Delta)$ be random variables t.v.i. $\calX \times \calT \times \bset$ where $\calX \subseteq \Reals^p$ and $\calT \subseteq \NNReals$.
Let $\calS \subseteq \distrT$ be a convex set of distributions on $\calT$ and let $\calR \subseteq \Reals$.
Then,

* The *probabilistic survival task* is the problem of predicting a conditional distribution over the positive Reals and is specified by $g: \calX \rightarrow \calS$.
* The *deterministic survival task* is the problem of predicting a continuous value in the positive Reals and is specified by $g: \calX \rightarrow \calT$.
* The *survival ranking task* is specified by predicting a continuous ranking in the Reals and is specified by $g: \calX \rightarrow \calR$.

:::
::::

Any other survival prediction type falls within one of the tasks in @cnj-task-surv, for example predicting log-survival time is the deterministic task and predicting prognostic index or linear predictor is the ranking task.
Removing the separation between the prognostic index and ranking prediction types is due to them both making predictions over the Reals; their mathematical difference lies in interpretation only.

> In this book, unless otherwise specified, the term *survival task*, will be used to refer to the probabilistic survival task.\footnote{These definitions are given in the most general case where the time variable is over $\NNReals$. In practice, all models instead assume time is over $\PReals$ and any death at $T_i = 0$ is set to $T_i = \epsilon$ for some very small $\epsilon \in \PReals$. Analogously for the discrete survival task. This assumption may not reflect reality as a patient could die at the study start however models cannot typically include this information in training.}

Reviewer: ```suggestion
In this book, unless otherwise specified, the term *survival task*, will be used to refer to the probabilistic survival task.
As a final note, these definitions are given in the most general case where the time variable is over $\NNReals$.
In practice, all models instead assume time is over $\PReals$ and so a subject $i$ that experiences an outcome at $0$ is either deleted or their outcome time is set to $T_i = \epsilon$ for some very small $\epsilon \in \PReals$. 
```

Should we talk about discrete modelling here? And/or discrete-time survival
